### PR TITLE
[feat]: [PL-26773]: Remove FF AUDIT_TRAIL_ENHANCEMENT as its GA

### PR DESCRIPTION
**Pull request by shashank-harness**
## Describe your changes
Removed FF as its GA

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37799)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37799)

**Review by ujjawal-harness — APPROVED** [review]
:lgtm:

> Reviewed 5 of 6 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799#-:-NCZ5vpFB2HYP_UhA3dL:bnfp4nl)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/security/AuthRuleFilter.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public void filter(ContainerRequestContext requestContext) {
       if (isEmpty(accountId)) {
         accountId = apiKeyService.getAccountIdFromApiKey(requestContext.getHeaderString("X-Api-Key"));
       }
-      if (isNotEmpty(accountId) && featureFlagService.isEnabled(FeatureName.AUDIT_TRAIL_ENHANCEMENT, accountId)) {
+      if (isNotEmpty(accountId)) {
```
Comment (meenakshi-harness) [inline 400-rest/src/main/java/software/wings/security/AuthRuleFilter.java:226]: Should we make this else for previous if ?

**Review by meenakshi-harness — DISMISSED** [review]
> Reviewed 5 of 6 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by ujjawal-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799#-:-NCxVbxs0HIQDcbTgFxa:bnfp4nl)* status: Some of the checks are too old.

**Review by ujjawal-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799#-:-NCxnjxTAXVTyzbH-no0:bnfp4nl)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by pbarapatre10 — APPROVED** [review]
:lgtm:

> Reviewed 4 of 6 files at r1, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799#-:-NCxntLrAA01XWaSpDBl:bnfp4nl)* status: :shipit: complete! ✓

**Review by kapil-harness — APPROVED** [review]
:lgtm:

> Reviewed 4 of 6 files at r1, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37799#-:-NCxrVjl7YWU-GyR91se:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**